cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/runtime/spi/runtimejob/TetheringRuntimeJobManager.java

```diff
@@ -106,10 +106,12 @@ public void launch(RuntimeJobInfo runtimeJobInfo) throws Exception {
     publishToControlChannel(message);
   }
 
+  /**
+  * This method should be used carefully as currently it always returns UNKNOWN as RuntimeJobStatus
+  * */
   @Override
   public Optional<RuntimeJobDetail> getDetail(ProgramRunInfo programRunInfo) {
-    // TODO: CDAP-18739 - pull job status instead of always treating it as RUNNING
-    return Optional.of(new RuntimeJobDetail(programRunInfo, RuntimeJobStatus.RUNNING));
+    return Optional.of(new RuntimeJobDetail(programRunInfo, RuntimeJobStatus.UNKNOWN));
   }
 
   @Override
```

Comment:
this is also a fake method, can we deprecate this in the RuntimeJobManager SPI? Looks like it's not used anywhere.

Then we can remove the todo comment here too

Reply:
this is being used for the Dataproc case in getStatus().
The corresponding Dataproc API returns a meaningful status.

Reply:
I don't see any usages when I check with my IDE. Which method in which class is calling this?

Reply:
In RemoteExecutionTwillController, we are planning to do - 

```

 while ((status = Retries.callWithRetries(
        remoteProcessController::getStatus, retryStrategy, Exception.class::isInstance)) == RuntimeJobStatus.RUNNING) {

```

This getStatus is as follows in RuntimeJobRemoteProcessController - 

```
  @Override
  public RuntimeJobStatus getStatus() throws Exception {
    try (RuntimeJobManager runtimeJobManager = runtimeJobManagerSupplier.get()) {
      return runtimeJobManager.getDetail(programRunInfo)
        .map(RuntimeJobDetail::getStatus)
        .orElse(RuntimeJobStatus.UNKNOWN);
    }
  }

```

The getDetail in case of TetheringRuntimeJobManager is 

```
  @Override
  public Optional<RuntimeJobDetail> getDetail(ProgramRunInfo programRunInfo) {
    return Optional.of(new RuntimeJobDetail(programRunInfo, RuntimeJobStatus.UNKNOWN));
  }

```

but for DataprocRuntimeJobManager is - 

```
  @Override
  public Optional<RuntimeJobDetail> getDetail(ProgramRunInfo programRunInfo) throws Exception {
    String jobId = getJobId(programRunInfo);

    try {
      Job job = getJobControllerClient().getJob(GetJobRequest.newBuilder()
                                                  .setProjectId(projectId)
                                                  .setRegion(region)
                                                  .setJobId(jobId)
                                                  .build());
      return Optional.of(new RuntimeJobDetail(getProgramRunInfo(job), getRuntimeJobStatus(job)));
    } catch (ApiException e) {
      if (e.getStatusCode().getCode() != StatusCode.Code.NOT_FOUND) {
        throw new Exception(String.format("Error while getting details for job %s on cluster %s.",
                                          jobId, clusterName), e);
      }
      // Status is not found if job is finished or manually deleted by the user
      LOG.debug("Dataproc job {} does not exist in project {}, region {}.", jobId, projectId, region);
    }
    return Optional.empty();
  }

```

So, I was wondering if we can deprecate it without affecting the Dataproc use case. As it does return meaningful values in the Dataproc case.

Reply:
Deprecating the list() method as discussed offline. 
Sorry for the confusion, I thought the getDetail() was being highlighted in the git comment.